### PR TITLE
Fix email notif utils + update new email address

### PR DESCRIPTION
**Pull request by noah-solomon**
## Overview

Email notifications were broken, so I used the [Sendgrid Quick Start](https://docs.sendgrid.com/for-developers/sending-email/quickstart-python) to fix. My guess is that the Sendgrid API has changed since this project was started. I also added the course name and number to the email that gets sent out to users.

## Changes Made

- Updated `.envrc` template based on recent changes & alphabetized 
- Added a filler `COURSE_NAME_NUM` to the email notif template, which gets replaced upon email send
- Send emails even if user is on mobile (in addition to push notification), not just for web users
- Refactor `src/app/coursegrab/notifications/push_notifications.py` to use `os.environ` instead of `environ` for clarity
- Update email sender to new address based on Sendgrid changes, `coursegrab-noreply@cornellappdev.com`
- Added Coursegrab logo to our email by making a dummy gmail account with same email address

## Test Coverage

Tested locally with success

## Next Steps

Potentially replace Sendgrid with another vendor (i.e. Amazon SES, Mailjet) so we cand send more emails for free

## Screenshots
image

### CODE REVIEW

**Review by katesliang — APPROVED** [review]
lptm (looks prod to me)! 👍

File: src/app/coursegrab/notifications/push_notifications.py [hunk 1]
```diff
@@ -27,13 +27,13 @@
 
 # Initialize SendGrid client
 try:
-    sendgrid_client = SendGridAPIClient(environ.get("SENDGRID_API_KEY"))
+    sendgrid_client = SendGridAPIClient(os.environ.get("SENDGRID_API_KEY"))
```
Comment (katesliang) [inline src/app/coursegrab/notifications/push_notifications.py:30]: very small nit but why do you do os.environ[...] everywhere else and os.environ.get here

Comment (snajima) [reply src/app/coursegrab/notifications/push_notifications.py:30]: +1

File: src/app/coursegrab/notifications/push_notifications.py [hunk 2]
```diff
@@ -27,13 +27,13 @@
...
 except:
     print("Error initializing SendGrid")
 
 # Initialize email body
 try:
-    f = open("./src/app/coursegrab/notifications/message.html", "r")
+    f = open(os.path.join(os.path.dirname(__file__), 'message.html'), "r")
```
Comment (snajima) [inline src/app/coursegrab/notifications/push_notifications.py:36]: lol why is the first string in single quotes and the second string not

**Review by snajima — APPROVED** [review]
great work noah 🎉 😍

**Pull request merged**